File: be2/app/models/faces.py
from config.db_helper import database  # koneksi database async
from datetime import datetime

async def get_all_faces():
    query = "SELECT name, embedding FROM faces"
    return await database.fetch_all(query=query)

async def insert_face(name: str, embedding: str):
    query = """
        INSERT INTO faces (name, embedding, created_at)
        VALUES (:name, :embedding, :created_at)
    """
    try:
        values = {
            "name": name,
            "embedding": embedding,
            "created_at": datetime.utcnow()
        }
        return await database.execute(query=query, values=values)
    except Exception as e:
        print("DB Insert Error:", e)
        raise

File: be3/src/models/faces.js
const conn = require('../config/DBHelper');

const getUserFaceData = () => {
    const QUERY = "SELECT name, embeddings FROM users";
    return conn.execute(QUERY);
}

const insertUsers = async (name, email, phone, embeddings) => {
    const QUERY = "INSERT INTO users (name, email, phone, embeddings, created_at) VALUES (?, ?, ?, ?, NOW())";
    try {
        const [result] = await conn.execute(QUERY, [name, email, phone, embeddings]);
        console.log("Insert result:", result);
        return result;
    } catch (error) {
        console.error("DB Insert Error:", error);
        throw error;
    }
};
module.exports = {
    getUserFaceData,
    insertUsers
}